src/DependencyEmitter/BasicDependencyEmitter.php

```diff
@@ -33,6 +33,42 @@ public function supportsParser(AstParserInterface $astParser)
         return $astParser instanceof NikicPhpParser;
     }
 
+    public function applyDependencies(
```

Comment:
please don't move and change functions, hard to review. i dont see any real benefit from moving this function, from my point of view it would be better to not rearrange it.